### ISSUE TITLE
ethproxy login does not accept worker name on some pools (0.14.rc9 and 0.15.dev7)

### ISSUE THREAD

**Issue by cnilsecure** [ethereum-mining/ethminer#1079]
I would join all the following topic to the this problem (#1070 ,  #1062)
for some reason both 0.14.rc9 and latest 0.15.X can not login to ethproxy pools:
  m  04:10:43|debug-ori|  ethminer 0.15.0.dev7-13+commit.d2e4ecd4
  m  04:10:43|debug-ori|  Build: linux / release
  ℹ  04:10:43|debug-ori|  Found suitable OpenCL device [ Ellesmere ] with 4281331712  bytes of GPU memory
  ℹ  04:10:43|debug-ori|  Selected pool us.clo.epool.io:8008
  m  04:10:43|debug-ori|  not-connected
  ℹ  04:10:43|stratum |  Trying 50.21.183.73:8008 ...
  ℹ  04:10:43|stratum |  Connected to us.clo.epool.io  [50.21.183.73:8008]
  ℹ  04:10:43|stratum |  Spinning up miners...
 cl  04:10:43|cl-0    |  No work. Pause for 3 s.
 cl  04:10:43|cl-1    |  No work. Pause for 3 s.
 cl  04:10:43|cl-2    |  No work. Pause for 3 s.
 cl  04:10:43|cl-3    |  No work. Pause for 3 s.
 cl  04:10:43|cl-4    |  No work. Pause for 3 s.
 cl  04:10:43|cl-5    |  No work. Pause for 3 s.
 cl  04:10:43|cl-6    |  No work. Pause for 3 s.
 cl  04:10:43|cl-7    |  No work. Pause for 3 s.
 cl  04:10:43|cl-8    |  No work. Pause for 3 s.
  ℹ  04:10:44|stratum |  Could not login to ethproxy server
  ℹ  04:10:44|stratum |  Disconnected from us.clo.epool.io  [50.21.183.73:8008]
  ℹ  04:10:44|stratum |  Shutting down miners...
 cl  04:10:46|cl-1    |  No work. Pause for 3 s.
  ✘  04:10:46|cl-0    |  OpenCL Error: clFinish: CL_INVALID_COMMAND_QUEUE (-36)
 cl  04:10:46|cl-2    |  No work. Pause for 3 s.
 cl  04:10:46|cl-3    |  No work. Pause for 3 s.
 cl  04:10:46|cl-4    |  No work. Pause for 3 s.
 cl  04:10:46|cl-5    |  No work. Pause for 3 s.
 cl  04:10:46|cl-6    |  No work. Pause for 3 s.
 cl  04:10:46|cl-7    |  No work. Pause for 3 s.
 cl  04:10:46|cl-8    |  No work. Pause for 3 s.
  ℹ  04:10:48|debug-ori|  Shutting down...
  ℹ  04:10:48|debug-ori|  Shutting down miners...
 ✘  04:10:49|cl-1    |  OpenCL Error: clFinish: CL_INVALID_COMMAND_QUEUE (-36)

any pool with ethproxy for some reason will not login
tried with eth-sh.poolgpu.com:8008 , etc1.91pool.com:8008 etc...

**Comment by AndreaLanfranchi** [comment]
Please if possible build latest releases from source. Problem has been 
addressed and solved but binary releases have not been published yet


Sent from mobile. Apologies for brevity and typos.

In data 7 maggio 2018 22:17:40 cnilsecure <notifications@github.com> ha 
scritto:

**Comment by cnilsecure** [comment]
This is build from lastest source.

**Comment by AndreaLanfranchi** [comment]
Meant latest source from branch release 0.14
Latest 0.15 has not been committed yes
Sorry for the inconveniences

Sent from mobile. Apologies for brevity and typos.

In data 7 maggio 2018 22:27:35 cnilsecure <notifications@github.com> ha 
scritto:

**Comment by jean-m-cyr** [comment]
PR #1080 Addresses this issue

**Comment by AndreaLanfranchi** [comment]
The assumption "ethproxy stops working" is apparently wrong. Just added some debug output

Using dwarfpool (ethproxy) using -P stratum1+tcp://$eth_address.BG01R02@eth-eu.dwarfpool.com:8008 i get.

```
  ℹ  07:22:22|stratum |  Connected to eth-eu.dwarfpool.com  [164.132.140.111:8008]
  ℹ  07:22:22|stratum |  Spinning up miners...
  ℹ  07:22:22|cuda-0  |  No work. Pause for 3 s.
  ℹ  07:22:22|cuda-1  |  No work. Pause for 3 s.
  ℹ  07:22:22|cuda-2  |  No work. Pause for 3 s.
  ℹ  07:22:22|cuda-4  |  No work. Pause for 3 s.
  ℹ  07:22:22|cuda-3  |  No work. Pause for 3 s.
  ℹ  07:22:22|cuda-5  |  No work. Pause for 3 s.
  ✘  07:22:22|stratum |  {
	"id" : 1,
	"method" : "eth_submitLogin",
	"params" : 
	[
		"<address>.BG01R02"
	],
	"worker" : "BG01R02"
}
  ℹ  07:22:22|stratum |  Logged in to eth-proxy server
  ✘  07:22:22|stratum |  {
	"id" : 5,
	"method" : "eth_getWork",
	"params" : []
}
  ℹ  07:22:22|stratum |  New pool difficulty:  2.00 gigahashes 
  ℹ  07:22:22|stratum |  New job #9b4c53d2…   eth-eu.dwarfpool.com [164.132.140.111:8008]
  ℹ  07:22:25|cuda-0  |  Initialising miner 0
  ℹ  07:22:25|cuda-2  |  Initialising miner 2
```

Which is perfectly valid.

Using pool evidenced by @cnilsecure with same -P argument I get

```
  ℹ  07:19:51|stratum |  Connected to us.clo.epool.io  [50.21.183.73:8008]
  ℹ  07:19:51|stratum |  Spinning up miners...
  ℹ  07:19:51|cuda-0  |  No work. Pause for 3 s.
  ℹ  07:19:51|cuda-1  |  No work. Pause for 3 s.
  ℹ  07:19:51|cuda-2  |  No work. Pause for 3 s.
  ℹ  07:19:51|cuda-3  |  No work. Pause for 3 s.
  ℹ  07:19:51|cuda-4  |  No work. Pause for 3 s.
  ℹ  07:19:51|cuda-5  |  No work. Pause for 3 s.
  ✘  07:19:51|stratum |  {
	"id" : 1,
	"method" : "eth_submitLogin",
	"params" : 
	[
		"<address>.BG01R02"
	],
	"worker" : "BG01R02"
}
  ℹ  07:19:52|stratum |  Could not login to ethproxy server
  ✘  07:19:52|stratum |  {
	"error" : 
	{
		"code" : -1,
		"message" : "Invalid login"
	},
	"id" : 1,
	"jsonrpc" : "2.0",
	"result" : null
}
```

**Comment by AndreaLanfranchi** [comment]
Solved in 0.15.dev8 and 0.14
